### PR TITLE
Minor refactor

### PR DESCRIPTION
**Pull request by theofidry**
Minor refactor to prepare #11.

Nor major changes, just broke down one bit to make it a bit easier to read.

### CODE REVIEW

File: src/BinCommand.php [hunk 1]
```diff
@@ -33,57 +34,85 @@ protected function configure()
      */
     public function execute(InputInterface $input, OutputInterface $output)
     {
-        $this->resetComposers();
+        $this->resetComposers($application = $this->getApplication());
```
Comment (theofidry) [inline src/BinCommand.php]: one avantage of making this `resetComposers()` stateless (i.e. passing the application), is that it can be validated via the typehint

File: src/BinCommand.php [hunk 2]
```diff
@@ -33,57 +34,85 @@ protected function configure()
...
-            $binRoot = $binVendorRoot.'/'.$binNamespace;
-            if (!file_exists($binRoot)) {
-                mkdir($binRoot, 0777, true);
-            }
-
-            $this->chdir($binRoot);
-
-            return $this->getApplication()->doRun($input, $output);
-        }
+        return ('all' !== $binNamespace)
```
Comment (theofidry) [inline src/BinCommand.php]: gist of the refactor: I find it easier to read in the sense that it conveys more what it does

File: src/BinCommand.php [hunk 3]
```diff
@@ -33,57 +34,85 @@ protected function configure()
...
+     * @return int
+     */
+    private function executeAllNamespaces(ComposerApplication $application, $binVendorRoot, InputInterface $input, OutputInterface $output)
+    {
         $binRoots = glob($binVendorRoot.'/*', GLOB_ONLYDIR);
         if (empty($binRoots)) {
             $this->getIO()->writeError('<warning>Couldn\'t find any bin namespace.</warning>');
 
-            return;
+            return 1;
```
Comment (theofidry) [inline src/BinCommand.php]: before nothing was returned, but in this case isn't it better to return `1` to say that an error occurred?

File: src/BinCommand.php [hunk 4]
```diff
@@ -33,57 +34,85 @@ protected function configure()
...
-            $this->resetComposers();
+            $this->resetComposers($application);
         }
 
         return min($exitCode, 255);
     }
 
     /**
+     * @param ComposerApplication $application
+     * @param string              $namespace
```
Comment (theofidry) [inline src/BinCommand.php]: most of the phpdoc is useless here, might be worth to remove the one unused (`$application`, `$input` and `$output`), `$namespace` and the return are still useful though

File: src/BinCommand.php [hunk 5]
```diff
@@ -33,57 +34,85 @@ protected function configure()
...
         return true;
     }
 
-    private function resetComposers()
+    /**
+     * Resets all Composer references in the application.
+     *
+     * @param ComposerApplication $application
+     */
+    private function resetComposers(ComposerApplication $application)
```
Comment (theofidry) [inline src/BinCommand.php]: btw why is this needed?

Comment (bamarni) [inline src/BinCommand.php]: There were some issues when changing directories and a composer application is loaded somewhere, IIRC it's mostly due to relative configs which were resolved with outdated full path, this makes sure there is a clean composer before each namespace command (cf. here for instance https://github.com/composer/composer/blob/22100b21eef2bb36fdbb47b23ddb69d2fab75fc3/src/Composer/Config.php#L34).

**Comment by bamarni** [comment]
thanks
